From e636b910d96cf9382722073fbdbd0110ea523c92 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 02 Sep 2022 21:04:32 +0300
Subject: [PATCH] [SW-102303] gaudi3: add hw rr config efuse bit api

Add api to read hw rr config (bit 16) . This is in preparation
to implement range registers in rom in case of failure in
hw to configure the range registers during boot.

Change-Id: Ic24cbc2d3ce33c9529f2a7f18c89678fbca6fdf0
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 8c47a6b..96e7530 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -515,6 +515,14 @@
 	security_smi_access(SMI_EN_VAL);
 }
 
+int security_efuse_hw_rr_config(void)
+{
+	/* check efuse for hw rr config */
+	uint32_t config_fuse = arcboot_rom_global.sec_data.efuse_sys_cfg;
+
+	return !!(config_fuse & EFUSE_SYSCFG_RR_MASK);
+}
+
 void security_disable_twi_if(void)
 {
 	security_smi_access(SMI_DIS_VAL);
